docs/extensions/network.md

```diff
@@ -0,0 +1,81 @@
+# Gardener Network Extension
+
+Gardener is an open-source project that provides a nested user model. Basically, there are two types of services provided by Gardener to its users:
+
+- Managed: users only request a Kubernetes cluster (Clusters-as-a-Service)
+- Hosted: users utilize Gardener to provide their own managed version of Kubernetes (Cluster-Provisioner-as-a-service)
+
+For the first set of users, the choice of network plugin might not be so important, however, for the second class of users (i.e., Hosted) it is important to be able to customize networking based on their needs.
+
+Furthermore, Gardener provisions clusters on different cloud-providers with different networking requirements. For example, Azure does not support Calico Networking [1], this leads to the introduction of manual exceptions in static add-on charts which is error prune and can lead to failures during upgrades.
+
+Finally, every provider is different, and thus the network always needs to adapt to the infrastructure needs to provider better performance. Consistency does not necessarily lie in the implementation but in the interface.
+
+## Motivation
+
+Prior to the `Network Extensibility` concept, Gardener followed a mono network-plugin support model (i.e., Calico). Although this seemed to be the more stable approach, it did not completely reflect the real use-case.
```

Comment:
```suggestion
Prior to the `Network Extensibility` concept, Gardener followed a mono network-plugin support model (i.e., Calico only). Although this seemed to be the more stable approach, it did not completely reflect the real use-case.
```

Reply:
I don't understand the second sentence. Why does this "seem to be the more stable approach"?

Reply:
The intention here is to say, it was thought that it is easier to maintain and control.

Reply:
Maybe you could re-formulate this. "more stable" sounds like what we are now doing is less stable.